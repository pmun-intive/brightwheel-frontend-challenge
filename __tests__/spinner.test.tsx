import { render, screen } from "@testing-library/react";
import Spinner from "@components/icon/spinner";

describe("Spinners", () => {
  it("renders a spinner svg with accessible name", () => {
    render(<Spinner />);

    const spinIcon = screen.getByRole("icon");

    expect(spinIcon).toHaveAccessibleName();
    expect(spinIcon).toMatchSnapshot();
  });

  it("renders a spinner svg with props width and height", () => {
    render(<Spinner height="250px" width="250px" />);

    const spinIcon = screen.getByRole("icon");

    expect(spinIcon).toHaveAttribute("width", "250px");
    expect(spinIcon).toHaveAttribute("height", "250px");
  });
});
