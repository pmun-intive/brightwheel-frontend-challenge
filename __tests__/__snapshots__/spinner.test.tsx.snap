// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinners renders a spinner svg with accessible name 1`] = `
<svg
  aria-label="An circle like spinning icon indicating some waiting for the information to be shown"
  height="100px"
  preserveAspectRatio="xMidYMid"
  role="icon"
  style="margin: auto; background: none; display: block; shape-rendering: auto;"
  viewBox="0 0 100 100"
  width="100px"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <circle
    cx="50"
    cy="50"
    fill="none"
    r="32"
    stroke="#ffffff"
    stroke-dasharray="50.26548245743669 50.26548245743669"
    stroke-linecap="round"
    stroke-width="8"
  >
    <animatetransform
      attributeName="transform"
      dur="1s"
      keyTimes="0;1"
      repeatCount="indefinite"
      type="rotate"
      values="0 50 50;360 50 50"
    />
  </circle>
</svg>
`;
